lms/djangoapps/certificates/models.py

```diff
@@ -464,6 +473,15 @@ def save(self, *args, **kwargs):  # pylint: disable=signature-differs
         The COURSE_CERT_AWARDED signal helps determine if a Program Certificate can be awarded to a learner in the
         Credentials IDA.
         """
+        try:
```

Comment:
Now I was able to configure my devstack correctly so that I could generate certs via:
- self generated by the user
- generated by the instructor because the learner had enough grades
- generate by the instructor based on the allowlist

When I did this and tried to load the progress page, it failed due to a recursion stack limit. I don't really think that it is due to this, because reverting the code back to the previous master did not fix it, however it got me thinking if a better place to trigger this filter would be https://github.com/openedx/edx-platform/blob/8070dd9be000da3fac86be19b5f0274f3d073630/lms/djangoapps/certificates/generation_handler.py#L88?

This way we are doing the filter while the task is still in the sync process, before it's sent to the async task.

Reply:
Maybe what you experienced has anything to do with this [PR](https://github.com/openedx/edx-platform/pull/29792)? In fact, it's in my backlog for review. If you want to review it go ahead! Given that you already replicated the error.

I believe you're right! That would be a better place to trigger the filter to avoid any shenanigans with async tasks. I'll move it to there and run some tests.

Reply:
I tested:

1. self generated by the user
2. generated by the instructor because the learner had a passing grade (through the command line, I'm not sure which method you used)
3. generated by the instructor based on the allowlist

And 1st the error you mentioned happened but disappeared after applying the PR I said.

What do you think? For method #1 we should modify how it's displayed in the template, and for method #3 it just doesn't say the "Certificate generated" date in the instructor panel as before I believe